src/extra/gfx/camera.cpp

```diff
@@ -0,0 +1,187 @@
+#include <foundation.hpp>
+#include <params.hpp>
+#include <linalg_shim.hpp>
+#include <gfx/math.hpp>
+#include <gfx/window.hpp>
+#include <gfx/fmt.hpp>
+#include "../gfx.hpp"
+
+using namespace shards;
+
+namespace gfx {
+
+struct FreeCameraShard : public gfx::BaseConsumer {
```

Comment:
The name is unclear. FreeFloating might be a good fit, but perhaps there is a better name, preferably another single word to use.

Reply:
MovableCamera maybe?